styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -753,9 +755,33 @@ private void processPodUpdate(String podName, WorkflowInstance instance) {
       emitPodEvents(pod, runState.get());
     }
 
+    private void reconnect() {
+      LOG.info("Re-establishing pod watcher");
+
+      closeWatch();
```

Comment:
This is not necessarily needed because fabric8 force closes it anyway. But to be on the safe side, we do it again and it is idempotent.